File: solutions/haskell/darts/1/src/Darts.hs
module Darts (score) where

magnitude :: Float -> Float -> Float
magnitude x y = sqrt((x ** 2) + (y ** 2))

score :: Float -> Float -> Int
score x y 
  | mag > 10 = 0
  | mag > 5 = 1
  | mag > 1 = 5
  | otherwise = 10
  where mag = magnitude x y 

File: solutions/c/darts/1/darts.c
#include "darts.h"
#include <math.h>


double magnitude(coordinate_t position) {
  double x_squared = pow(position.x, 2.0);
  double y_squared = pow(position.y, 2.0);

  return sqrt(x_squared + y_squared);
}

unsigned int score(coordinate_t position) {
  double length = magnitude(position);
  if (length > 10.0) {
    return 0;
  } else if (length <= 1.0 ) {
    return 10;
  } else if (length <= 5.0) {
    return 5;
  } else {
    return 1;
  }
}
